.github/actions/build-backend/action.yml

```diff
@@ -30,11 +44,32 @@ runs:
       shell: bash
       run: ./gradlew package -x fatjar -Pshowtests
 
+    - name: Build Prime Router test report
+      working-directory: prime-router
+      run: ./gradlew jacocoTestReport -x fatjar
+      shell: bash
+
     - name: Tar router files
       working-directory: prime-router
       shell: bash
       run: tar -czvf prime-router-build.tar.gz build
 
+    - name: SonarCloud Scan
+      if: inputs.GITHUB_TOKEN != '' && inputs.SONAR_TOKEN != ''
+      uses: sonarsource/sonarcloud-github-action@db501078e936e4b4c8773d1bb949ba9ddb7b6b6a
+      env:
+        GITHUB_TOKEN: ${{ inputs.GITHUB_TOKEN }}
+        SONAR_TOKEN: ${{ inputs.SONAR_TOKEN }}
+      with:
+        args: >
+          -Dsonar.cpd.exclusions=prime-router/src/testIntegration/**,prime-router/src/main/kotlin/cli/tests/**
+          -Dsonar.coverage.exclusions=prime-router/src/test/**,prime-router/src/testIntegration/**,prime-router/src/main/kotlin/cli/tests/**
+          -Dsonar.sources=prime-router/src
```

Comment:
I've found that if we don't include both the frontend and backend source location in the sonarcloud call, the missing source results get deleted from sonarclouds results data.